packages/terra-abstract-modal/src/AbstractModal.jsx

```diff
@@ -87,38 +82,76 @@ class AbstractModal extends React.Component {
   constructor() {
     super();
     this.handleKeydown = this.handleKeydown.bind(this);
+    this.setAbstractModalRef = this.setAbstractModalRef.bind(this);
+    this.showModalDomUpdates = this.showModalDomUpdates.bind(this);
+    this.hideModalDomUpdates = this.hideModalDomUpdates.bind(this);
   }
 
   componentDidMount() {
     document.addEventListener('keydown', this.handleKeydown);
+
+    if (this.props.isOpen) {
+      this.showModalDomUpdates();
+    }
   }
 
   componentDidUpdate(prevProps) {
-    // When the Modal is no longer in focus, it should no longer listen to the keydown event to handle the Escape key.
-    // Otherwise, the Modal would also get closed when the intention for pressing the Escape key is to close a popup inside the modal.
-    if (!this.props.isFocused && prevProps.isFocused) {
-      document.removeEventListener('keydown', this.handleKeydown);
-    } else if (this.props.isFocused && !prevProps.isFocused) {
-      document.addEventListener('keydown', this.handleKeydown);
+    if (this.props.isOpen) {
+      if (!prevProps.isOpen) {
+        this.showModalDomUpdates();
+      }
+    } else if (prevProps.isOpen) {
+      this.hideModalDomUpdates();
     }
   }
 
   componentWillUnmount() {
     document.removeEventListener('keydown', this.handleKeydown);
-    this.toggleVisuallyHiddenMainDocument('false');
+    this.hideModalDomUpdates();
   }
 
-  handleKeydown(e) {
-    if (e.keyCode === KeyCode.KEY_ESCAPE && this.props.isOpen && this.props.closeOnEsc && this.props.isFocused) {
-      this.props.onRequestClose();
+  setAbstractModalRef(node) {
+    this.AbstractModalRef = node;
+  }
+
+  showModalDomUpdates() {
+    const mainDocumentElement = document.querySelector(this.props.rootSelector);
+
+    if (mainDocumentElement) {
+      mainDocumentElement.setAttribute('aria-hidden', 'true');
+      mainDocumentElement.setAttribute('inert', '');
+      // Shift focus to modal when opened
+      this.AbstractModalRef.focus();
     }
   }
 
-  toggleVisuallyHiddenMainDocument(hiddenValue) {
+  hideModalDomUpdates() {
     const mainDocumentElement = document.querySelector(this.props.rootSelector);
 
     if (mainDocumentElement) {
-      mainDocumentElement.setAttribute('aria-hidden', hiddenValue);
+      mainDocumentElement.setAttribute('aria-hidden', 'false');
+      mainDocumentElement.removeAttribute('inert');
+    }
+
+    if (this.modalTrigger) {
+      // Shift focus back to element that was last focused prior to opening the modal
+      this.modalTrigger.focus();
+    }
+
+    setTimeout(() => {
+      if (this.modalTrigger) {
+        // Shift focus back to element that was last focused prior to opening the modal
+        this.modalTrigger.focus();
+      }
+    }, 100); // Allows inert processing to finish before shifting focus back
```

Comment:
Am I seeing double with lines 136 and 142? Are these both necessary?

Reply:
Ah, meant to remove lines 136-138.

Reply:
Updated in this commit: https://github.com/cerner/terra-framework/pull/608/commits/e8719e68c7b13e7a911edc17bae6a73df49e1f7e